pandas/tests/series/test_rank.py

```diff
@@ -134,22 +134,27 @@ def test_rank_categorical(self):
         assert_series_equal(ordered.rank(), exp)
         assert_series_equal(ordered.rank(ascending=False), exp_desc)
 
-        # Unordered categoricals should be ranked as objects
+        # See #19560
+        error_msg = ("pandas.core.algorithms.rank "
+                     "not supported for unordered "
+                     "non-numeric data")
+
+        # Ranking unordered categorials depreciated per #19560
         unordered = Series(['first', 'second', 'third', 'fourth',
                             'fifth', 'sixth']).astype(
             CategoricalDtype(categories=['first', 'second', 'third',
                                          'fourth', 'fifth', 'sixth'],
                              ordered=False))
-        exp_unordered = Series([2., 4., 6., 3., 1., 5.])
-        res = unordered.rank()
-        assert_series_equal(res, exp_unordered)
+
+        with tm.assert_raises_regex(ValueError, error_msg):
+            unordered.rank()
 
         unordered1 = Series(
             [1, 2, 3, 4, 5, 6],
         ).astype(CategoricalDtype([1, 2, 3, 4, 5, 6], False))
-        exp_unordered1 = Series([1., 2., 3., 4., 5., 6.])
-        res1 = unordered1.rank()
-        assert_series_equal(res1, exp_unordered1)
+
+        # Won't raise ValueError because entries not objects.
+        unordered1.rank()
```

Comment:
Is this right? Certainly the example passes the "eye test" but if the Categorical is not ordered what semantics are we using for ranking?